### ISSUE TITLE
player.configure({textDisplayFactory:...}) crashes in player.unload() with a release build

### ISSUE THREAD

**Issue by aptly-io** [google/shaka-player#1187]
What is the correct way for using the textDisplayFactory field in the player's configure structure?

We're using a custom TTML text renderer/displayer. Displaying subtitles works fine. 

After switching from the `shaka-player.compiled.debug.js` to `shaka-player.compiled.js` we notice a crash in the players' `unload()`. This does not happen with the debug build.

```
player.js:1825 Uncaught (in promise) TypeError: a.w.o is not a function
    at dh (player.js:1825)







    at mh (player.js:1859)
    at U.<anonymous> (player.js:1145)
    at <anonymous>
```

**Have you read the [FAQ](https://goo.gl/JE1Sy5) and checked for duplicate issues**:
Yes

**What version of Shaka Player are you using**:
master branch

**Can you reproduce the issue with our latest release version**:
Yes

**Can you reproduce the issue with the latest code from `master`**:
Yes

**Are you using the demo app or your own custom app**:
My own based on the shaka tutorial "Basic Usage"

**If custom app, can you reproduce the issue using our demo app**:
No but there's sample code attached to reproduce: 
Just dump the 2 attached files in the `$ROOT/demo` directory 
`./build/all.py` , start `http-server -c-1 --cors="*" -p12345` from the root
Now load `fm_demo.html` with Chrome

**What browser and OS are you using**:
Chrome (also seeing this on the chromecast)

**What are the manifest and license server URIs**:
*(you can send the URIs to <shaka-player-issues@google.com> instead, but please use GitHub and the template for the rest)*

**What did you do?**
Using the shaka-player.compiled.js

  player.configure({
    textDisplayFactory: function() {
      return new Displayer();
    }
  });

  setTimeout(function() {
    console.log('### UNLOADING');
    player.unload();                  // crash happens here (or in the `.catch()`)
  }, 8000);

**What did you expect to happen?**
Graceful unloading without leaving the player in an inconsistant state.

**What actually happened?**
Seeing this crash in the chrome's developer console, leaving the player in an inconsistant state

**Comment by aptly-io** [comment]
[fm_demo.txt](https://github.com/google/shaka-player/files/1551653/fm_demo.txt)
[fm_app.txt](https://github.com/google/shaka-player/files/1551654/fm_app.txt)

**Comment by joeyparrish** [comment]
Sounds like a bug in the way we declared the class.  I expect it's TextDisplayer.destroy() that is the issue.  Let me do some testing and see if I can confirm that.

**Comment by joeyparrish** [comment]
It appears to be the only external plugin class that implements `shaka.util.IDestroyable`, and we left out the declaration of `destroy()` in the externs.  This caused the compiler to rename the method inside the compiled library.

**Comment by aptly-io** [comment]
@joeyparrish Thank you for your fix!

**Comment by joeyparrish** [comment]
We are always happy to help.

**Comment by joeyparrish** [comment]
The fix has been cherry-picked for v2.2.9.
